tests/test_2024_11_26.py

```diff
@@ -0,0 +1,379 @@
+    assert pml_period_duration_months_after == 3
+    assert pml_spend_limit_after == pml_spend_limit_after_expected
+    assert interface.AllowedRecipientRegistry(pml_allowed_recipients_registry).isUnderSpendableBalance(pmlSpendableBalanceAfter, 3)
+    assert pmlSpendableBalanceAfter == pml_spend_limit_after_expected
+    limit_test(easy_track,
+               int(pmlSpendableBalanceAfter / (10**18)) * 10**6,
+               pml_trusted_caller_acc,
+               pml_top_up_evm_script_factory,
+               pml_multisig_acc,
+               stranger,
+               interface.Usdc(USDC_TOKEN),
+               2_000_000 * 10**6
+    )
+
+    # Item 3
+    stonks_steth_budget_limit_after, stonks_steth_period_duration_months_after = interface.AllowedRecipientRegistry(stonks_steth_allowed_recipients_registry).getLimitParameters()
+    assert stonks_steth_budget_limit_after == stonks_steth_budget_after_expected
+    assert stonks_steth_period_duration_months_after == 6
+
+    # Item 4
+    already_spent_amount_after, _, stonks_steth_period_start_after, stonks_steth_period_end_after = stonks_steth_allowed_recipients_registry.getPeriodState()
+    assert stonks_steth_period_start_after == stonks_steth_period_start_exptected
+    assert stonks_steth_period_end_after == stonks_steth_period_end_exptected
+    assert already_spent_amount_after == stonks_steth_new_spent_amount_expected
+    stonks_steth_spendable_balance_after = interface.AllowedRecipientRegistry(stonks_steth_allowed_recipients_registry).spendableBalance()
+    assert stonks_steth_spendable_balance_after == stonks_steth_budget_after_expected
+    limit_test(easy_track,
+               stonks_steth_spendable_balance_after,
+               stonks_steth_trusted_caller,
+               stonks_steth_top_up_evm_script_factory,
+               stonks_steth_contract,
+               stranger,
+               stETH_token,
+               5_000 * 10 ** 18
+    )
+
+    # Item 5
+    simply_staking_data_after = nor.getNodeOperator(simply_staking_id, True)
+    assert simply_staking_new_reward_address == simply_staking_data_after[rewardAddressIndex]
+    assert simply_staking_data_before[nameIndex] == simply_staking_data_after[nameIndex]
+    assert simply_staking_data_before[activeIndex] == simply_staking_data_after[activeIndex]
+    assert simply_staking_data_before[stakingLimitIndex] == simply_staking_data_after[stakingLimitIndex]
+    assert simply_staking_data_before[stoppedValidatorsIndex] == simply_staking_data_after[stoppedValidatorsIndex]
+
+
+    # events
+    display_voting_events(vote_tx)
+    evs = group_voting_events(vote_tx)
+
+    metadata = find_metadata_by_vote_id(vote_id)
+    assert get_lido_vote_cid_from_str(metadata) == "bafkreia2qh6xvoowgwukqfyyer2zz266e2jifxovnddgqawruhe2g5asgi"
+
+    assert count_vote_items_by_events(vote_tx, contracts.voting) == 5, "Incorrect voting items count"
+
+    validate_set_limit_parameter_event(
```

Comment:
if there's a way to also check on what particular contract (registry) this event was emitted that would be perfect

Reply:
here and after